### PR TITLE
[cssom] Test serialization of media queries

### PR DESCRIPTION
**Pull request by jugglinmike**
Some discrepancies between browsers became apparent while writing gh-17221

- Chrome passes all 12 tests
- Firefox fails two tests (feature de-duplication and feature sorting)
- Safari fails all 12 tests (maybe a bit unfairly--all due to its insertion of an extra U+0020)

@tabatkins How are duplicated value-specifying features intended to be interpreted? For instance,

```css
@media print ans (max-width: 23px) and (max-width: 45px) {}
```

### CODE REVIEW

**Comment by tabatkins** [comment]
> How are duplicated value-specifying features intended to be interpreted?

They're just separate MQs, evaluated independently. In your example, the MQ will be true if the width is <= 23px (as that passes both of the 'max-width' MQs) and false otherwise.

**Comment by jugglinmike** [comment]
@TabAtkins would you mind reviewing this for me?

**Comment by jugglinmike** [comment]
@dbaron would you mind reviewing this for me?

File: css/cssom/serialize-media-rule.html [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+test(function(t) {
+  var sheet = makeSheet(t);
+  sheet.insertRule('@media screen and (color) and (color) {}');
+
+  assert_equals(sheet.cssRules.length, 1);
+  assert_equals(
+    sheet.cssRules[0].cssText,
+    '@media screen and (color) {\n}'
+  );
+}, 'features - de-duplicatoin');
```
Comment (dbaron) [inline css/cssom/serialize-media-rule.html]: please fix spelling of duplication

**Review by dbaron — COMMENTED** [review]
So I had a look at the [failures](https://staging.wpt.fyi/results/css/cssom/serialize-media-rule.html?label=pr_head&max-count=1&pr=17236) and I guess I'm a little concerned about the first of the Firefox ones.

I don't see anything [in the spec](https://drafts.csswg.org/cssom/#serialize-a-media-query) that requires de-duplication.  While the second *example* in that section suggests de-duplication is required, I don't see any normative text in the spec requiring it.  Do you know of any issues filed about this?

Otherwise I think this looks fine.

(I don't think "lexicographical" (rather than "lexicographic") is a word, though... but the spec uses it, so it's ok for the test to match.)

**Comment by dbaron** [comment]
Also, it would probably be good to test the [example in your opening comment](https://github.com/web-platform-tests/wpt/pull/17236#issue-286273658) that it doesn't de-duplicate.

**Comment by jugglinmike** [comment]
Thanks, @dbaron!

> I don't see anything [in the spec](https://drafts.csswg.org/cssom/#serialize-a-media-query) that requires de-duplication.  While the second *example* in that section suggests de-duplication is required, I don't see any normative text in the spec requiring it.

I think you're right. [Parse a media query list](https://drafts.csswg.org/cssom/#parse-a-media-query-list) uses [media query list](https://drafts.csswg.org/mediaqueries-4/#media-query-list), which itself uses [parse a comma-separated list of component values](https://drafts.csswg.org/css-syntax-3/#parse-a-comma-separated-list-of-component-values). That final algorithm allows duplication, and I don't see any restrictions imposed by the callers.

I've commented out the test so that future reviewers can see this omission is deliberate (and potentially temporary).

> Do you know of any issues filed about this?

[This was reported three years ago.](https://github.com/w3c/csswg-drafts/issues/533) I've left a comment to point to this new test.

> Also, it would probably be good to test the [example in your opening comment](https://github.com/web-platform-tests/wpt/pull/17236#issue-286273658) that it doesn't de-duplicate.

Definitely. Done.

**Comment by jugglinmike** [comment]
This pull request was being rejected by Taskcluster [because it was based on a relatively old commit](https://github.com/web-platform-tests/wpt/issues/17286#issuecomment-501208404). I've rebased the branch on the latest revision of `master` as Sam suggested in that issue, and that corrected the problem.

[The original version of this branch is available here.](https://github.com/web-platform-tests/wpt/compare/master...bocoup:css-media-query-serialize-orig)

**Review by dbaron — APPROVED** [review]
I wish I could see the diff between the last PR and this one, but it looks like the changes I expected to see based on a pretty quick skim...

**Comment by dbaron** [comment]
(er, I guess it was in the fixup commit, and that does look right)

**Comment by jugglinmike** [comment]
Sorry about that. I typically try to avoid rebasing during a review.

**Pull request merged**